lazy_tensor_core/lazy_tensor_core/csrc/device.cpp

```diff
@@ -4,20 +4,16 @@
 #include <c10/util/Optional.h>
 
 namespace torch_lazy_tensors {
-namespace {
-
-thread_local c10::optional<Device> g_current_device;
```

Comment:
it seems like you have moved the thread-local device from the bridge to the tensor_impl, but the basic semantics (one global setting per thread) remains the same.  I think that's fine, just confirming i understood correctly.

Reply:
Yea, your are right!